src/components/timers/Countdown.tsx

```diff
@@ -1,3 +1,183 @@
-const Countdown = () => null;
+import { useEffect, useRef, useState } from 'react';
+import { STATUS } from '../../utils/constants';
+import type { StatusType } from '../../utils/constants';
+import { DisplayForText, DisplayForTime, convertToSeconds } from '../../utils/helpers';
+import { Button, Buttons, Input, Inputs, SupportText, TimeDisplay, Timer, TimerContainer, TimerTitle } from '../../utils/styles';
+
+const Countdown = () => {
+    const [timeMinInput, setTimeMinInput] = useState('');
+    const [timeSecInput, setTimeSecInput] = useState('');
+
+    const [status, setStatus] = useState<StatusType>(STATUS.INITIAL);
+    const [secondsRemaining, setSecondsRemaining] = useState(0);
+
+    const secondsOnTimer = secondsRemaining % 60;
+    const minutesRemaining = (secondsRemaining - secondsOnTimer) / 60;
+    const minutesOnTimer = minutesRemaining % 60;
+    const hoursOnTimer = (minutesRemaining - minutesOnTimer) / 60;
+
+    const totalSeconds = convertToSeconds(timeMinInput, timeSecInput);
+
+    const startStopCountdown = () => {
+        if (Number.isNaN(totalSeconds) || (timeMinInput === '' && timeSecInput === '') || totalSeconds <= 0) {
+            alert('Please enter a valid time.');
+        } else if (totalSeconds > 3600) {
+            alert('Friendly caution: excercise over an hour can lead to overtraining. Please enter a time under an hour.');
+        } else {
+            if (status !== STATUS.STARTED) {
+                if (secondsRemaining === 0) {
+                    setSecondsRemaining(totalSeconds);
+                }
+                setStatus(STATUS.STARTED);
+            } else {
+                setStatus(STATUS.STOPPED);
+            }
+        }
+    };
+
+    const resetCountdown = () => {
+        setStatus(STATUS.STOPPED);
+        const totalSeconds = convertToSeconds(timeMinInput, timeSecInput);
+        setSecondsRemaining(totalSeconds);
+    };
+
+    const fastforwardCountdown = () => {
+        setStatus(STATUS.FASTFORWARDED);
+        setSecondsRemaining(0);
+        if (intervalRef.current !== null) {
+            clearInterval(intervalRef.current);
+            intervalRef.current = null;
+        }
+    };
+
+    const initialCountdown = () => {
+        setStatus(STATUS.INITIAL);
+    };
+
+    const intervalRef = useRef<number | null>();
+
+    useEffect(() => {
```

Comment:
Great job handling the interval/use effect here! I see some repeat logic in the other timers - this is a great candidate for a custom hook. You could consider refactoring that for assignment 2